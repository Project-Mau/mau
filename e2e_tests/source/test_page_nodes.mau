= Test page nodes

= Header 1

This file tests some of the page nodes like header.

// This is a comment and will not appear in the output.

== Header 2

More headers.

===== Header 5

I'm skipping some steps here.

////
This is a multiline
comment.
All these lines will disappear.
////

---

Just above this paragraph there should be a horizontal line.

---

A bit more fun with styles, just to check that everything is working.

You can have _*strong and empashized*_ text.

You can also apply styles to _*`verbatim`*_.

But verbatim will `_*preserve*_` them.

You can use _single *markers.

But you \_need\_ to escape pairs.

Even though you can escape \_only one_ of the two.

If you have \_more than two\_ it's better to just \_escape\_ all of them.

Oh, this is valid for `verbatim as well.

---

This is an unordered list

* Item 1
* Item 2
** Item 2.1
** Item 2.2
* Item 3

This is an ordered list

# Item 1
# Item 2
## Item 2.1
## Item 2.2
# Item 3

Finally, let's include an image.

.A beautiful image
<< image:"https://images.unsplash.com/photo-1534106474077-f9e9c6f5a47c?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1740&q=80"
