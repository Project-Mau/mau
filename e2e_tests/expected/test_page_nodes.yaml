data:
  args: []
  content:
  - data:
      anchor: test-page-nodes-cca6
      args: []
      kwargs: {}
      level: 1
      tags: []
      type: header
      value: Test page nodes
  - data:
      anchor: header-1-1aa5
      args: []
      kwargs: {}
      level: 1
      tags: []
      type: header
      value: Header 1
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: This file tests some of the page nodes like header.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      anchor: header-2-871b
      args: []
      kwargs: {}
      level: 2
      tags: []
      type: header
      value: Header 2
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: More headers.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      anchor: header-5-3193
      args: []
      kwargs: {}
      level: 5
      tags: []
      type: header
      value: Header 5
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: I'm skipping some steps here.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      kwargs: {}
      tags: []
      type: horizontal_rule
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: Just above this paragraph there should be a horizontal line.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      kwargs: {}
      tags: []
      type: horizontal_rule
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: A bit more fun with styles, just to check that everything is
                working.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'You can have '
          - data:
              content:
                data:
                  content:
                  - data:
                      content:
                        data:
                          content:
                          - data:
                              type: text
                              value: strong and empashized
                          type: sentence
                      type: style
                      value: star
                  type: sentence
              type: style
              value: underscore
          - data:
              type: text
              value: ' text.'
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'You can also apply styles to '
          - data:
              content:
                data:
                  content:
                  - data:
                      content:
                        data:
                          content:
                          - data:
                              type: verbatim
                              value: verbatim
                          type: sentence
                      type: style
                      value: star
                  type: sentence
              type: style
              value: underscore
          - data:
              type: text
              value: .
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'But verbatim will '
          - data:
              type: verbatim
              value: _*preserve*_
          - data:
              type: text
              value: ' them.'
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: You can use _single *markers.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: But you _need_ to escape pairs.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: Even though you can escape _only one_ of the two.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: If you have _more than two_ it's better to just _escape_ all
                of them.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: Oh, this is valid for `verbatim as well.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      kwargs: {}
      tags: []
      type: horizontal_rule
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: This is an unordered list
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      items:
      - data:
          content:
            data:
              content:
              - data:
                  type: text
                  value: Item 1
              type: sentence
          level: 1
          type: list_item
      - data:
          content:
            data:
              content:
              - data:
                  type: text
                  value: Item 2
              - data:
                  args: []
                  items:
                  - data:
                      content:
                        data:
                          content:
                          - data:
                              type: text
                              value: Item 2.1
                          type: sentence
                      level: 2
                      type: list_item
                  - data:
                      content:
                        data:
                          content:
                          - data:
                              type: text
                              value: Item 2.2
                          type: sentence
                      level: 2
                      type: list_item
                  kwargs: {}
                  main_node: false
                  ordered: false
                  tags: []
                  type: list
              type: sentence
          level: 1
          type: list_item
      - data:
          content:
            data:
              content:
              - data:
                  type: text
                  value: Item 3
              type: sentence
          level: 1
          type: list_item
      kwargs: {}
      main_node: true
      ordered: false
      tags: []
      type: list
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: This is an ordered list
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      items:
      - data:
          content:
            data:
              content:
              - data:
                  type: text
                  value: Item 1
              type: sentence
          level: 1
          type: list_item
      - data:
          content:
            data:
              content:
              - data:
                  type: text
                  value: Item 2
              - data:
                  args: []
                  items:
                  - data:
                      content:
                        data:
                          content:
                          - data:
                              type: text
                              value: Item 2.1
                          type: sentence
                      level: 2
                      type: list_item
                  - data:
                      content:
                        data:
                          content:
                          - data:
                              type: text
                              value: Item 2.2
                          type: sentence
                      level: 2
                      type: list_item
                  kwargs: {}
                  main_node: false
                  ordered: true
                  tags: []
                  type: list
              type: sentence
          level: 1
          type: list_item
      - data:
          content:
            data:
              content:
              - data:
                  type: text
                  value: Item 3
              type: sentence
          level: 1
          type: list_item
      kwargs: {}
      main_node: true
      ordered: true
      tags: []
      type: list
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: Finally, let's include an image.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      alt_text: null
      args: []
      classes: null
      kwargs: {}
      tags: []
      title:
        data:
          content:
          - data:
              type: text
              value: A beautiful image
          type: sentence
      type: content_image
      uri: https://images.unsplash.com/photo-1534106474077-f9e9c6f5a47c?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1740&q=80
  kwargs: {}
  tags: []
  type: document

